api/app/GraphQL/Validators/Mutation/UpdateWorkExperienceValidator.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+namespace App\GraphQL\Validators\Mutation;
+
+use App\Enums\EmploymentCategory;
+use App\Enums\GovEmploymentType;
+use App\Models\WorkExperience;
+use Illuminate\Validation\Rule;
+use Nuwave\Lighthouse\Validation\Validator;
+
+final class UpdateWorkExperienceValidator extends Validator
+{
+    /**
+     * Return the validation rules.
+     *
+     * @return array<string, array<mixed>>
+     */
+    public function rules(): array
+    {
+        $workExperienceModel = WorkExperience::findOrFail($this->arg('id'));
+
+        // new arguments take priority, otherwise fallback to model value
+        $employmentCategory = $this->arg('workExperience.employmentCategory') ?? $workExperienceModel->employment_category;
+        $extSizeOfOrganization = $this->arg('workExperience.extSizeOfOrganization') ?? $workExperienceModel->ext_size_of_organization;
+        $extRoleSeniority = $this->arg('workExperience.extRoleSeniority') ?? $workExperienceModel->ext_role_seniority;
+        $govEmploymentType = $this->arg('workExperience.govEmploymentType') ?? $workExperienceModel->gov_employment_type;
+        $govPositionType = $this->arg('workExperience.govPositionType') ?? $workExperienceModel->gov_position_type;
+        $govContractStartDate = $this->arg('workExperience.govContractStartDate') ?? $workExperienceModel->gov_contract_start_date;
```

Comment:
Will delete some of these unused ones after some reviewing is done

Reply:
I dunno, this doesn't smell quite right to me.  Laravel validation (and Lighthouse) is all about validation input but really you want to validate the final state of the model.  This seems very clunky though.  I think I'll ask on Slack if anyone has ideas.

Reply:
So uh, what do we want?

Reply:
Decision in scrum: don't query existing model, just expect that all the fields to check exist in the input.

Reply:
Aye aye, reusing the validator then in https://github.com/GCTC-NTGC/gc-digital-talent/pull/11947/commits/688a7a92e1aeb127ca32f955c276b728fce723ed